nixos/modules/services/networking/nftables.nix

```
@@ -185,6 +185,19 @@ in
           can be loaded using "nft -f".  The ruleset is updated atomically.
         '';
     };
+
+    networking.nftables.rulesetFileFlatten = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Flatten the rulesetFile instead of including it. It is useful when you want to
+        apply the preCheckRuleset to the ruleset file.
+
+        This option will use "builtins.readFile" rather than "include" to handle the file.
+        So it is expected that the file can be accessed from the sanbox environment.
```

```suggestion
        Use `builtins.readFile` rather than `include` to handle {option}`networking.nftables.rulesetFile`. It is useful when you want to apply {option}`networking.nftables.preCheckRuleset` to {option}`networking.nftables.rulesetFile`.

        ::: {.note}
        It is expected that {option}`networking.nftables.rulesetFile` can be accessed from the build sandbox.
        :::
```

https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#syntax